resources/prosody-plugins/mod_filter_iq_rayo.lua

```diff
@@ -75,13 +76,20 @@ module:hook("pre-iq/full", function(event)
             -- if current user is not allowed, but was granted moderation by a user
             -- that is allowed by its features we want to allow it
             local is_granting_session_allowed = false;
-            if (session.granted_jitsi_meet_context_features) then
+            if session.granted_jitsi_meet_context_features then
                 is_granting_session_allowed = is_feature_allowed(session.granted_jitsi_meet_context_features, feature);
             end
 
-            if (token == nil
-                or roomName == nil
-                or not token_util:verify_room(session, room_jid_match_rewrite(roomName))
+            local room_real_jid = room_jid_match_rewrite(roomName);
+
+            if not session.jitsi_meet_context_features and not session.granted_jitsi_meet_context_features then
```

Comment:
Same as above. This is non-trivial code, and can be extracted to a central place. It would be much easier to read and verify if we just had `local is_allowed = is_feature_allowed(...)` here

Reply:
Same as the other one, there is no common place in this project to extract features and know is the participant a moderator.